src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -3163,6 +3197,24 @@ private boolean writeBatchData(TDSWriter tdsWriter) throws SQLServerException {
             // No more data available, return false so we do not execute any more batches.
             if (!goToNextRow())
                 return false;
+            
+            
+            if (insertRowByRow) {
+                // read response gotten from goToNextRow()
+                ((SQLServerResultSet) sourceResultSet).getTDSReader().readPacket();
+
+                // Create and send the initial command for bulk copy ("INSERT BULK ...").
+                tdsWriter = command.startRequest(TDS.PKT_QUERY);
+                String bulkCmd = createInsertBulkCommand(tdsWriter);
+                tdsWriter.writeString(bulkCmd);
```

Comment:
the thing is, we have to reset the `tdsWriter` after `goToNextRow()`, and `createInsertBulkCommand` requires the new `tdsWriter`, so, it has to be here, otherwise it fails when column is encrypted